File: leetcode/array/1151_Minimum_Swaps_to_Group_All_1s_Together/codes/sliding.py
from typing import List

class Solution:
    def minSwaps(self, data: List[int]) -> int:
        ones = sum(data)
        if ones <= 1:
            return 0

        current_ones = sum(data[:ones])
        max_ones_in_window = current_ones

        for i in range(ones, len(data)):
            current_ones += data[i] - data[i - ones]
            max_ones_in_window = max(max_ones_in_window, current_ones)

        return ones - max_ones_in_window

File: leetcode/array/1151_Minimum_Swaps_to_Group_All_1s_Together/codes/sliding.c
int minSwaps(int* data, int dataSize) {
    int ones = 0;
    for (int i = 0; i < dataSize; i++) {
        ones += data[i];
    }
    if (ones <= 1) return 0;

    int currentOnes = 0;
    for (int i = 0; i < ones; i++) {
        currentOnes += data[i];
    }

    int maxOnes = currentOnes;
    for (int i = ones; i < dataSize; i++) {
        currentOnes += data[i] - data[i - ones];
        if (currentOnes > maxOnes) maxOnes = currentOnes;
    }

    return ones - maxOnes;
}
